### ISSUE TITLE
No compile error when `conditionExpression` and `expression` are used together

### ISSUE THREAD

**Issue by JanecekPetr** [mapstruct/mapstruct#2794]
Using MapStruct 1.5.0.RC1, imagine the following (or a matching class hierarchy, doesn't matter)

```java
// domain
interface Result {}
record Success(Instant timestamp) implements Result {}
record Failure(String reason) implements Result {}

// public API
record ResultApi(boolean success, Instant timestamp, String reason) {}
```

and the following mapper:
```java
@Mapping(target = "success", expression = "java(result instanceof Success)")
@Mapping(target = "timestamp",
    expression = "java(result instanceof Success ? ((Success) result).getTimestamp() : null)")
@Mapping(target = "reason",
    expression = "java(result instanceof Failure ? ((Failure) result).getReason() : null)")
ResultApi resultToResultApi(Result result);
```

This works. I tried changing the mapper to
```java
@Mapping(target = "success", expression = "java(result instanceof Success)")
@Mapping(target = "timestamp",
    conditionExpression = "java(result instanceof Success)",
    expression = "java(((Success) result).getTimestamp())")
@Mapping(target = "reason",
    conditionExpression = "java(result instanceof Failure)",
    expression = "((Failure) result).getReason())")
ResultApi resultToResultApi(Result result);
```

however the conditionals never appeared in the generated code.

**Comment by JanecekPetr** [comment]
Oh, I am *officially* blind. I did not see that line, sorry. Indeed, that's my bad. Obviously it would be nice if I got a compile-time error instead, but this is working as documented, indeed.

**Comment by filiphr** [comment]
Thanks for reporting this @JanecekPetr, it should have been a compile time error. I'll keep this issue open so we can fix this in the final 1.5 release.
